src/helics/core/CommonCore.cpp

```diff
@@ -3744,4 +3744,21 @@ void CommonCore::processMessageFilter (ActionMessage &cmd)
     }
 }
 
+    const std::string &CommonCore::getInterfaceInfo(interface_handle handle) const
+    {
+        auto handleInfo = getHandleInfo (handle);
+        if (handleInfo != nullptr)
+        {
+            return handleInfo->interface_info;
+        }
+        return emptyStr;
+    }
+
+    void CommonCore::setInterfaceInfo(helics::interface_handle handle, std::string info) {
```

Comment:
And this would be the trouble spot?  
Here you need to use a modify function on handles directly like,  may need some checks as well to make sure handle is valid
```
handles.modify ([&](auto &hdls) { hdls.getHandleInfo (handle.baseValue ())->interface_info = info; });
```

Reply:
the handles is protected by a ordered_guarded object which only allows modifications through the modify function so you need to write a little lambda function that does the modifications